pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerInterceptorTest.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import org.apache.pulsar.broker.intercept.BrokerInterceptor;
+import org.apache.pulsar.common.api.proto.BaseCommand;
+import org.apache.pulsar.common.api.proto.CommandAck;
+import org.apache.pulsar.common.api.proto.MessageIdData;
+import org.apache.pulsar.common.protocol.Commands;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+
+public class BrokerInterceptorTest {
```

Comment:
There is an existing test `BrokerInterceptorTest` in package `org.apache.pulsar.broker.intercept`.

Reply:
@gaoran10  yes, but I want to mock the `handleAck` method, it's a protected method......

Reply:
@tjiuming If not able to use the existing `BrokerInterceptorTest`, could you please move the new test class under the `org.apache.pulsar.broker.intercept` package and use a different class name? Otherwise, the new test class most likely not be used in the future.

And the new test class should under the "broker" test group.

BTW, looks like it's easy to improve the existing test `testBeforeSendMessage` to cover the new change, send some messages, consume and ack them later, then check the interceptor works well.